components/button/floating-buttons.js

```diff
@@ -147,75 +165,98 @@ class FloatingButtons extends RtlMixin(LitElement) {
 		`;
 	}
 
-	_calcContainerPosition() {
-		this._floating = this._shouldFloat();
-		if (!this._floating || !this.offsetParent) {
-			return;
+	updated(changedProperties) {
+		if (changedProperties.has('alwaysFloat')) {
+			this._recalculateFloating();
 		}
+	}
 
-		const offsetParentBoundingRect = this.offsetParent.getBoundingClientRect();
-		const boundingRect = this.getBoundingClientRect();
+	async _calcContainerPosition() {
+		requestAnimationFrame(() => {
 
-		const offsetParentLeft = offsetParentBoundingRect.left;
-		const left = boundingRect.left;
-		const containerLeft = left - offsetParentLeft - 1;
-		if (containerLeft !== 0) {
-			// only update this if needed - needed for firefox
-			this._containerMarginLeft = `-${containerLeft}px`;
-		}
+			this._containerMarginLeft = '';
+			this._containerMarginRight = '';
+			this._innerContainerLeft = '';
+			this._innerContainerRight = '';
 
-		const offsetParentRight = offsetParentBoundingRect.right;
-		const right = boundingRect.right;
-		const containerRight = offsetParentRight - right - 1;
-		if (containerRight !== 0) {
-			this._containerMarginRight = `-${containerRight}px`;
-		}
+			if (!this._floating) return;
+
+			const boundingAncestor = this._getBoundingAncestor();
+			if (!boundingAncestor) return;
 
-		if (this.dir !== 'rtl') {
+			const offsetParentBoundingRect = boundingAncestor.getBoundingClientRect();
+			const boundingRect = this.getBoundingClientRect();
+
+			const offsetParentLeft = offsetParentBoundingRect.left;
+			const left = boundingRect.left;
+			const containerLeft = left - offsetParentLeft - 1;
 			if (containerLeft !== 0) {
-				this._innerContainerLeft = `${containerLeft}px`;
+				// only update this if needed - needed for firefox
+				this._containerMarginLeft = `-${containerLeft}px`;
 			}
-		} else {
-			this._innerContainerRight = `${containerRight}px`;
-		}
+
+			const offsetParentRight = offsetParentBoundingRect.right;
+			const right = boundingRect.right;
+			const containerRight = offsetParentRight - right - 1;
+			if (containerRight !== 0) {
+				this._containerMarginRight = `-${containerRight}px`;
+			}
+
+			if (this.dir !== 'rtl') {
+				if (containerLeft !== 0) {
+					this._innerContainerLeft = `${containerLeft}px`;
+				}
+			} else {
+				this._innerContainerRight = `${containerRight}px`;
+			}
+
+		});
 	}
 
-	_shouldFloat() {
-		if (this.alwaysFloat) {
-			return true;
+	_getBoundingAncestor() {
```

Comment:
When searching, we pick wither the bounding ancestor (`overflow: scroll`) or offsetParent (`position: relative`) that we see first by traveling up the DOM.